### PR TITLE
PR: Enclosing quotes of sys.argv clearing instruction are now consistent

### LINKED ISSUE #3987: Errors running or importing code because sys.argv is empty
Hi,
Upgraded to spyder 3.1. and tried to import tensorflow and got these errors on the ipython console
```
import tensorflow
Traceback (most recent call last):

  File "<ipython-input-1-a649b509054f>", line 1, in <module>
    import tensorflow

  File "/usr/local/lib/python2.7/dist-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import *

  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/__init__.py", line 100, in <module>
    from tensorflow.python.platform import app

  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 23, in <module>
    from tensorflow.python.platform import flags

  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/flags.py", line 25, in <module>
    _global_parser = _argparse.ArgumentParser()

  File "/usr/lib/python2.7/argparse.py", line 1586, in __init__
    prog = _os.path.basename(_sys.argv[0])

IndexError: list index out of range
```

It was working before in spyder3.0.2. tensorflow and ipython (5.1.0) versions have not changed. It also works on spyder's python console tab:

```
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow
>>> tensorflow.__version__
'0.12.1'
```
OS is Ubuntu 16.04 64 bit
P.S same issue with spyder git master.

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #3987

### CODE REVIEW

**Comment by ccordoba12** [comment]
@andfoy, did you test this change? Is it working fine for you now?

**Comment by ccordoba12** [comment]
And don't worry the failing tests in Travis, that's not related to your work :-)

**Comment by andfoy** [comment]
@ccordoba12 I tested them indeed!

**Comment by ccordoba12** [comment]
Ok, great!

**Comment by ccordoba12** [comment]
@goanpeca, could you instruct @andfoy on how to add a test for this?

**Comment by goanpeca** [comment]
lets have a hangout tomorrow please ping me

**Comment by ccordoba12** [comment]
@andfoy, please have the hangout with @goanpeca and add a test for this tomorrow. I plan to release 3.1.1 on Saturday, so that's fine :-)

**Comment by goanpeca** [comment]
@andfoy this is looking great, good job :-)

**Comment by andfoy** [comment]
@goanpeca Well, to initialize the client, it is necessary to initialize an IPython kernel, along to other variables only known to the plugin.

**Comment by goanpeca** [comment]
>@goanpeca Well, to initialize the client, it is necessary to initialize an IPython kernel, along to other variables only known to the plugin.

Ok, in that case @ccordoba12 we will need to rework that plugin logic, cause the ipython console (as a whole widget) should be self contained, no?

File: spyder/plugins/tests/test_argvclear.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    client = ipy.get_current_client()
+    with qtbot.waitSignal(client.shell_ready, timeout=6000) as blocker:
+        shell.silent_exec_method('import sys;A = len(sys.argv)')
+    len_argv = shell.get_value("A")
+    while len_argv is None:
+        shell.silent_exec_method('import sys;A = len(sys.argv)')
+        qtbot.keyClicks(client.get_control(), 'import sys;A = len(sys.argv)')
+        qtbot.keyPress(client.get_control(), Qt.Key_Return)
+        len_argv = shell.get_value("A")
+    assert len_argv > 0
```
Comment (goanpeca) [inline spyder/plugins/tests/test_argvclear.py]: Should this be an specific value for the test?

1 or 2 or exactly 1 or 2? that would make the test more robust

File: spyder/plugins/tests/test_argvclear.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    widget.show()
+    return qtbot, widget
+
+# Tests
+#-------------------------------
+def test_sys_argv_clear(botipython):
+    qtbot, ipy = botipython
+    shell = ipy.get_current_shellwidget()
+    client = ipy.get_current_client()
+    with qtbot.waitSignal(client.shell_ready, timeout=6000) as blocker:
```
Comment (goanpeca) [inline spyder/plugins/tests/test_argvclear.py]: For readability lets add an empty space before the `with`

File: spyder/plugins/tests/test_argvclear.py (hunk 1 above)
Comment (andfoy) [inline spyder/plugins/tests/test_argvclear.py]: Well, it should be exactly one

Comment (goanpeca) [inline spyder/plugins/tests/test_argvclear.py]: Then lets make it assert len_argv == 1

**Comment by ccordoba12** [comment]
> why did we use the plugin and not the client? (that is the main widget that the plugin uses)

What @andfoy did was the right thing to do because the plugin is the one in charge of starting kernels and assigning them to clients.

> we will need to rework that widget logic (in another PR in the future), cause the ipython console (as a whole widget) should be self contained, no?

Hard to do it because it needs to know about other plugins. But we could try to do it.

**Comment by ccordoba12** [comment]
Great good @andfoy! Thanks a lot for the IPython console tests :-)

I also took the opportunity to fix some other tests that were failing.

**Pull request merged**